### PR TITLE
MM-16265 MM-16507 - Migrate to bot accounts

### LINKED ISSUE #90: Migrate Jira plugin to use bot accounts
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Plugin: Jira" community channel](https://community.mattermost.com/core/channels/jira-plugin).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/) and our [Plugins Guide](https://developers.mattermost.com/extend/plugins/).

----

There are two stages for the issue:

1 - Migrate Jira plugin to use bot accounts.

For an example, here is a PR that migrated the Demo Plugin to use bot accounts with a helper function https://github.com/mattermost/mattermost-plugin-demo/pull/29

2 - Remove `User` Mattermost System Console setting in plugins.json.

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Migrate to bot accounts
- Moved command responses to ephemeral posts. To make this as painless as possible, refactored `responsef(...)` -> `p.responsef(header, ...)`. Was suggested by ChristopherS -- we don't need an iconURL if we send ephemeral posts.
- Removed UserName from the config. This eliminated a couple tests (if the EnsureBot fails, the plugin won't be running).
- Moved to 5.12
- Fixed the token issue (I know it's in the backlog of merges, but it was annoying :)

Looks like:
image

image

#### Links
- Fixes [MM-16265](https://mattermost.atlassian.net/browse/MM-16265)
- [MM-16507](https://mattermost.atlassian.net/browse/MM-16507)
- Fixes #90

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -489,14 +492,18 @@ func getCommand() *model.Command {
...
-		Username:     PluginMattermostUsername,
-		IconURL:      PluginIconURL,
-		Type:         model.POST_DEFAULT,
+func (p *Plugin) postCommandResponse(args *model.CommandArgs, text string) {
+	post := &model.Post{
+		UserId:    p.getUserID(),
+		ChannelId: args.ChannelId,
+		Message:   text,
 	}
+	_ = p.API.SendEphemeralPost(args.UserId, post)
```
Comment (levb) [inline server/command.go]: I always thought that CommandResponse was a weird layer. I 👍 this.

File: server/plugin.go [hunk 2]
```diff
@@ -99,10 +100,31 @@ func (p *Plugin) OnConfigurationChange() error {
...
+	p.updateConfig(func(conf *config) {
+		conf.botUserID = botUserID
+	})
+
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return errors.Wrap(err, "couldn't get bundle path")
+	}
+
+	profileImage, err := ioutil.ReadFile(filepath.Join(bundlePath, "assets", "profile.png"))
```
Comment (levb) [inline server/plugin.go]: Should put this in the `/public`? @hanzei would want it I think, and would be right. We didn't do it before only to be 5.10-compatible.

**Comment by jasonblais** [comment]
Tagging `Docs: Needed` label as we'll want to update docs prior to the release

File: server/plugin.go (hunk 2 above)
Comment (hanzei) [reply server/plugin.go]: No, this should not be in `/public`. `/public` is used to serve files via http. This file just gets read and then send via RPC. There is not need to expose this to the public.

This is fine the way it is.

**Pull request merged**